File: inventory-backend/src/main/java/com/ems/profile/model/StoreProfile.java
package com.ems.profile.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "store_profile")
public class StoreProfile {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String shopName;
    private String address;
    private String gstin;
    private String stateCode;
    private String phone;
    private String jurisdictionCourt;
}

File: inventory-frontend/src/lib/api/storeProfile.ts
import { apiClient } from "./client";

export type StoreProfile = {
  id: number;
  shopName: string;
  address: string;
  gstin: string;
  stateCode: string;
  phone: string;
  jurisdictionCourt: string;
};

export async function fetchStoreProfile(): Promise<StoreProfile> {
  const { data } = await apiClient.get<StoreProfile>("/store-profile");
  return data;
}

export async function updateStoreProfile(
  profile: Partial<StoreProfile>,
): Promise<StoreProfile> {
  const { data } = await apiClient.put<StoreProfile>("/store-profile", profile);
  return data;
}
